aepp/flowservice.py

```diff
@@ -461,21 +493,146 @@ def createFlow(self, obj: dict = None) -> dict:
                 requires following keys : name, flowSpec, sourceConnectionIds, targetConnectionIds, transformations, scheduleParams.
         """
         if obj is None:
-            raise Exception("Require a dictionary to create the flow")
-        if "name" not in obj.keys():
-            raise KeyError("missing 'name' parameter in the dictionary")
-        if "flowSpec" not in obj.keys():
-            raise KeyError("missing 'flowSpec' parameter in the dictionary")
-        if "sourceConnectionIds" not in obj.keys():
-            raise KeyError("missing 'sourceConnectionIds' parameter in the dictionary")
-        if "targetConnectionIds" not in obj.keys():
-            raise KeyError("missing 'targetConnectionIds' parameter in the dictionary")
+            if any(param is None for param in [name, source_connection_id, target_connection_id]):
+                raise KeyError("Require either obj or all of 'name', 'source_connection_id', 'target_connection_id'")
+            if schedule_frequency not in ("minute", "hour"):
+                raise ValueError("schedule frequency has to be either minute or hour")
+            obj = {
+                "name": name,
+                "flowSpec": {
+                    "id": flow_spec_id,
+                    "version": version
+                },
+                "sourceConnectionIds": [
+                    source_connection_id
+                ],
+                "targetConnectionIds": [
+                    target_connection_id
+                ],
+                "transformations": [],
+                "scheduleParams": {
```

Comment:
are these required params, what happens if these are not specified or the customer wants to run manually

Reply:
I am not aware if the scheduleFrequency is really required for every type of Flow. Like HTTP Ingestion per example. 
We may want to do it optional.

Reply:
Agree, good catch, I changed and made the params inside of `scheduleParams` optional (but from the docs under https://developer.adobe.com/experience-platform-apis/references/flow-service/#tag/Flows/operation/createFlow it seems the object `scheduleParams` is required so keeping this top-level field even if it ends up empty I think makes sense.